src/components/apply/form/application-form.tsx

```diff
@@ -1,4 +1,147 @@
-// TODO: finish form component
+import type { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+} from "~/components/ui/form";
+import { Textarea } from "~/components/ui/textarea";
+import { api } from "~/utils/api";
+import { useAutoSave } from "~/components/hooks/use-auto-save";
+import {
+  applicationSaveSchema,
+  applicationSubmitSchema,
+  basicsSaveSchema,
+} from "~/schemas/application";
+import { useState } from "react";
+import { applications } from "~/server/db/schema";
+
 export function ApplicationForm() {
-  return <></>;
+  const utils = api.useUtils();
+  const { data: defaultValues } = api.application.get.useQuery();
+  const { mutate } = api.application.save.useMutation({
+    onSuccess: () => {
+      return utils.application.get.invalidate();
+    },
+  });
+
+  const form = useForm<z.infer<typeof applicationSubmitSchema>>({
+    resolver: zodResolver(basicsSaveSchema),
+    mode: "onBlur",
+  });
+
+  useAutoSave(form, onSubmit);
```

Comment:
```suggestion
  useAutoSave(form, onSubmit, defaultValues);
```
so that the form initializes with the correct values